core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1860,16 +1860,20 @@ public void run() {
                   .set(LoadBalancer.ATTR_HEALTH_CHECKING_CONFIG, healthCheckingConfig)
                   .build();
             }
+            Attributes attributes = attrBuilder.build();
 
-            Status handleResult = helper.lb.tryAcceptResolvedAddresses(
+            boolean addressesAccepted = helper.lb.tryAcceptResolvedAddresses(
                 ResolvedAddresses.newBuilder()
                     .setAddresses(servers)
-                    .setAttributes(attrBuilder.build())
+                    .setAttributes(attributes)
                     .setLoadBalancingPolicyConfig(effectiveServiceConfig.getLoadBalancingConfig())
                     .build());
 
-            if (!handleResult.isOk()) {
-              handleErrorInSyncContext(handleResult.augmentDescription(resolver + " was used"));
+            if (!addressesAccepted) {
+              handleErrorInSyncContext(Status.UNAVAILABLE.withDescription(
```

Comment:
The problem is this will replace the useful error message the LB has with this unhelpful one. It also generates two calls to the error handler for one event. That's why I had mentioned just scheduling the exponential backoff (which will soon go away as well, moving into the NR).

Reply:
I changed it so it will call `scheduleExponentialBackOffInSyncContext()` and updated the related unit tests, but I'm a bit uncertain if this really is the behavior we want - could you please doublecheck the commit with the change before I merge this?